### PR TITLE
misc: align PyTorch Engine temprature with TurboMind

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

as titled

Hi @grimoire @lzhangzz @lvhan028 May you help review this? Thanks.

ref TurboMind
https://github.com/InternLM/lmdeploy/blob/c59a70413c3600fb22e683c46e085758272e4178/src/turbomind/kernels/sampling_penalty_kernels.cu#L149-L161

https://github.com/InternLM/lmdeploy/blob/c59a70413c3600fb22e683c46e085758272e4178/src/turbomind/kernels/sampling_penalty_kernels.cu#L75-L86

## Modification

as titled

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
In our use case, if algorithm colleagues use a model, they often use temperature 0. If we directly change it to 1 according to PyTorch's implementation, this is actually inconsistent with our expected usage. Seeing that in TurboMind, a very small value was added to avoid division by zero, I think this is a good solution. And we want the behavior of the PyTorch Engine and TurboMind in LMDeploy to be basically consistent.

File: lmdeploy/pytorch/messages.py [hunk 1]
```diff
@@ -72,7 +72,10 @@ def from_gen_config(self, gen_config: EngineGenerationConfig):
             logger.warning('`top_p` has to be a float > 0 and < 1'
                            f' but is {top_p}')
             top_p = 1.0
-        if temperature <= 0:
+        if temperature == 0:
```
Comment (lvhan028) [inline lmdeploy/pytorch/messages.py:75]: @grimoire what's the minimum temperature in torch engine?

Comment (grimoire) [reply lmdeploy/pytorch/messages.py:75]: https://github.com/InternLM/lmdeploy/blob/a06174f836882d853d4eb18519c2245c2a7eae8c/lmdeploy/pytorch/messages.py#L75-L78

zero or negative temperature would be set to 1

Comment (grimoire) [reply lmdeploy/pytorch/messages.py:75]: Only accept positive temperature.

Comment (zhyncs) [reply lmdeploy/pytorch/messages.py:75]: > Only accept positive temperature.

Users may use temperature 0 in the sampling parameter, and currently this modification meets this requirement.

Comment (grimoire) [reply lmdeploy/pytorch/messages.py:75]: Accept

Comment (zhyncs) [reply lmdeploy/pytorch/messages.py:75]: > Accept

cc @lvhan028

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**